src/ConstructiveSolidGeometry/VolumePrimitives/Box.jl

```diff
@@ -46,12 +46,48 @@ box:
 
 See also [Constructive Solid Geometry (CSG)](@ref).
 """
-@with_kw struct Box{T, CO} <: AbstractVolumePrimitive{T, CO}
-    hX::T = 1
-    hY::T = 1
-    hZ::T = 1
-    origin::CartesianPoint{T} = zero(CartesianPoint{T})
-    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+struct Box{T,CO} <: AbstractVolumePrimitive{T,CO}
+    hX::T
+    hY::T
+    hZ::T
+    
+    origin::CartesianPoint{T}
+    rotation::SMatrix{3,3,T,9}
+end
+
+#Type conversion happens here
+function Box{T}(CO,hX, hY, hZ, origin, rotation) where {T}
+    _hX = _csg_convert_args(T, hX)
+    _hY = _csg_convert_args(T, hY)
+    _hZ = _csg_convert_args(T, hZ)
+    Box{T,CO}(_hX, _hY, _hZ, origin, rotation)
+end
+
+#Type promotion happens here
+function Box(CO, hX::TX, hY::TY, hZ::TZ, origin::PT, rotation::ROT) where {TX, TY, TZ, PT, ROT}
+    eltypes = _csg_get_promoted_eltype.((TX, TY, TZ, PT, ROT))
+    T = promote_type(eltypes...)
+    Box{T}(CO,T(hX), T(hY), T(hZ), origin, rotation)
+end
+
+function Box(::Type{CO}=ClosedPrimitive;
+    hX = 1.0,
+    hY = 1.0,
+    hZ = 1.0,
+    origin = zero(CartesianPoint{Float64}), 
+    rotation = one(SMatrix{3, 3, Float64, 9})
+) where {CO}
+    Box(CO, hX, hY, hZ, origin, rotation)
+end
+
```

Comment:
Please implement the default parameters as `Int` and include `float` in the type promotion as done in #278.